src/components/widgets/Dropzone.js

```diff
@@ -0,0 +1,179 @@
+      return;
+    }
+
+    this.setState({
+      content: (
+        <div className="dropzone-container__message">
+          <p>
+            {_('Drop ') +
+              (this.props.fileType ? this.props.fileType : 'file') +
+              _(' to upload.')}
+          </p>
+
+          {this.props.fileType === 'image' ? (
+            <p>
+              {_('Supported formats are: ') +
+                this.validFiletypes[this.props.fileType] +
+                '.'}
+            </p>
+          ) : null}
+        </div>
+      ),
+    });
+  }
+
+  onLoad(e) {
+    const _ = this.props.localize;
+    const parsingError = (
+      <div className="dropzone-container__message">
+        <p>{_('Yikes! An error occurred while parsing this file.')}</p>
+        <p>
+          {this.props.fileType
+            ? _('Try again with a supported file format: ') +
+              this.validFiletypes[this.props.fileType] +
+              '.'
+            : _('Try again.')}
+        </p>
+      </div>
+    );
+
+    if (this.props.fileType === 'image') {
+      try {
+        this.props.onUpdate(e.target.result);
+        this.setState({
+          content: this.renderSuccess(e.target.result),
+        });
+      } catch (error) {
+        console.warn(error); // eslint-disable-line
+        this.setState({
+          content: parsingError,
+        });
+      }
+    }
+  }
+
+  onDrop(file) {
+    const _ = this.props.localize;
+    const reader = new FileReader();
+
+    const invalidFileTypeMessage = this.props.fileType ? (
+      <div className="dropzone-container__message">
+        <p>
+          {_(
+            `Yikes! This doesn't look like a valid ${
+              this.props.fileType
+            } to us. `
+          )}
+        </p>
+        <p>
+          {_('Try again with a ') +
+            this.validFiletypes[this.props.fileType] +
+            '.'}
+        </p>
+      </div>
+    ) : (
+      _('Unsupported file format!')
+    );
+
+    if (file.length > 1) {
+      this.setState({
+        content: (
+          <div className="dropzone-container__message">
+            <p>{_('Yikes! You can only upload one file at a time.')}</p>
+            <p>
+              {_(
+                'To upload multiple files, create multiple files and upload them individually.'
+              )}
+            </p>
+          </div>
+        ),
+      });
+      return;
+    }
+
+    this.setState({content: _('Loading...')});
+
+    reader.onload = e => this.onLoad(e);
+
+    if (this.props.fileType === 'image') {
+      if (
+        ['.jpeg', '.jpg', '.svg', '.png', '.gif', '.bmp', '.webp'].some(ext =>
+          file[0].name.endsWith(ext)
+        )
+      ) {
+        reader.readAsDataURL(file[0]);
+      } else {
+        this.setState({
+          content: invalidFileTypeMessage,
+        });
+      }
+    }
+  }
+
+  render() {
+    return (
+      <Drop
+        ref="dzone"
+        onDrop={this.onDrop}
+        disableClick={true}
```

Comment:
I would enable click, so that people can also use the file dialog instead of being required to drag.

Reply:
`/create` supports clicking on the word "upload" for this purpose, but really the whole area can be clickable and it works just as well, like I did here https://react-pivottable.js.org/